snapshots/GasTest.json

```diff
@@ -0,0 +1,26 @@
+{
+  "exactInput_multi_ERC20_to_ERC20_customCurve": "180726",
+  "exactInput_multi_ERC20_to_ERC20_hookData": "202072",
+  "exactInput_multi_ERC20_to_ERC20_nativeIntermediate": "187280",
+  "exactInput_multi_ERC20_to_ERC20_vanilla": "193451",
+  "exactInput_multi_ERC20_to_ETH_nativeOutput": "178659",
+  "exactInput_multi_ETH_to_ERC20_nativeInput": "177859",
+  "exactInput_single_ERC20_to_ERC20_customCurve": "120376",
+  "exactInput_single_ERC20_to_ERC20_encoded": "126837",
+  "exactInput_single_ERC20_to_ERC20_encoded_permit2": "167204",
+  "exactInput_single_ERC20_to_ERC20_hookData": "141726",
+  "exactInput_single_ERC20_to_ERC20_vanilla": "133113",
```

Comment:
```
  "exactInput_single_ERC20_to_ERC20_encoded": "126837",
  "exactInput_single_ERC20_to_ERC20_vanilla": "133113",
```
looks like we're losing 6k gas to `abi.encode(BaseData)` 🤔

Reply:
interesting. I am curious how much calldata size might differ between them